src/Sulu/Bundle/AdminBundle/Resources/js/components/ResourceLocator/ResourceLocator.js

```diff
@@ -0,0 +1,55 @@
+// @flow
+import React from 'react';
+import Input from '../Input';
+import resourceLocatorStyles from './resourceLocator.scss';
+
+type Props = {
+    value: string,
+    onChange: (value: string) => void,
+    mode: 'full' | 'leaf',
+};
+
+export default class ResourceLocator extends React.PureComponent<Props> {
+    static defaultProps = {
+        value: '/',
+    };
+
+    fixed: string = '';
+    changeable: string = '';
+
+    componentWillMount = () => {
+        const {value, mode} = this.props;
+        let parts;
+
+        switch (mode) {
+            case 'full':
+                this.fixed = '/';
+                this.changeable = value.substring(1);
+                break;
+            case 'leaf':
+                parts = value.split('/');
```

Comment:
Instead of outside the `switch` you could use `const` and instantiate the `parts` variable here

Reply:
Ok, so this seems to fail eslint because of this rule: https://github.com/lukeapage/eslint-plugin-switch-case/blob/master/docs/rules/no-case-curly.md

What's our stand on this? /cc @chirimoya

Reply:
Actually it seems to be this rule: https://eslint.org/docs/rules/no-case-declarations We've decided to deactivate this rule and declare this variable in the `case` statement.